uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderConcurrentOperationsTest.java

```diff
@@ -0,0 +1,82 @@
+package org.uberfire.java.nio.fs.jgit;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.*;
+
+public class JGitFileSystemProviderConcurrentOperationsTest extends AbstractTestInfra {
+
+    private Logger logger = LoggerFactory.getLogger(JGitFileSystemProviderConcurrentOperationsTest.class);
+
+    @Test
+    public void testConcurrentGitCreation() {
+
+        int threadCount = 2;
```

Comment:
In order to proper test those scenarios we should use byteman. But I'm more than happy to add this on a second PR.

Reply:
@ederign can we create an issue for this? So I can improve this in the next weeks?

Reply:
Could you please open a jira?